src/test/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/ParseIcebergColumnTreeTest.java

```diff
@@ -0,0 +1,243 @@
+    IcebergColumnJsonValuePair columnValuePair =
+        IcebergColumnJsonValuePair.from(recordNode.fields().next());
+
+    IcebergColumnTree modifiedTree = treeFactory.fromJson(columnValuePair);
+    // then
+    mergeTreeService.merge(alreadyExistingTree, modifiedTree);
+
+    String expected = expectedResult.replaceAll("/ +/g", " ");
+    Assertions.assertEquals(expected, typeBuilder.buildType(alreadyExistingTree));
+    Assertions.assertEquals("TESTSTRUCT", alreadyExistingTree.getColumnName());
+  }
+
+  static Stream<Arguments> mergeTestArguments() {
+    return Stream.of(
+        arguments(
+            "{\"type\":\"struct\",\"fields\":["
+                + "{\"id\":23,\"name\":\"k1\",\"required\":false,\"type\":\"int\"},"
+                + "{\"id\":24,\"name\":\"k2\",\"required\":false,\"type\":\"int\"}"
+                + "]}",
+            "{ \"testStruct\": { \"k1\" : 1, \"k2\" : 2, \"k3\" : 3 } }",
+            "OBJECT(k1 INT, k2 INT, k3 LONG)"),
+        arguments(
+            "{\"type\":\"struct\",\"fields\":["
+                + "{\"id\":23,\"name\":\"k1\",\"required\":false,\"type\":\"int\"},{\"id\":24,\"name\":\"k2\",\"required\":false,\"type\":\"int\"},"
+                + "  {\"id\":25,\"name\":\"nested_object\",\"required\":false,\"type\":{\"type\":\"struct\",\"fields\":["
+                + "    {\"id\":26,\"name\":\"nested_key1\",\"required\":false,\"type\":\"string\"},"
+                + "    {\"id\":27,\"name\":\"nested_key2\",\"required\":false,\"type\":\"string\"}"
+                + "]}}]}",
+            "{\"testStruct\" : {"
+                + "  \"k1\" : 1, "
+                + "  \"k2\" : 2, "
+                + "    \"nested_object\": { "
+                + "      \"nested_key1\" : \"string\", "
+                + "      \"nested_key2\" : \"blah\", "
+                + "      \"nested_object2\" : { "
+                + "        \"nested_key2\" : 23.5 "
+                + "    }}"
+                + "}}",
+            "OBJECT(k1 INT, k2 INT, nested_object OBJECT(nested_key1"
+                + " VARCHAR(16777216), nested_key2 VARCHAR(16777216), nested_object2"
+                + " OBJECT(nested_key2 DOUBLE)))"),
+        // ARRAY merge
+        arguments(
+            "{\"type\":\"list\",\"element-id\":23,\"element\":\"long\",\"element-required\":false}",
+            "{\"TESTSTRUCT\": [1,2,3] }",
+            "ARRAY(LONG)"),
+        arguments(
+            "{\"type\":\"list\",\"element-id\":1,\"element\":{"
+                + "\"type\":\"struct\",\"fields\":["
+                + "{\"id\":1,\"name\":\"primitive\",\"required\":true,\"type\":\"boolean\"}"
+                + "]},"
+                + "\"element-required\":true}",
+            "{\"TESTSTRUCT\": [ { \"primitive\" : true, \"new_field\" : 25878749237493287429348 }]"
+                + " }",
+            "ARRAY(OBJECT(primitive BOOLEAN, new_field LONG))"));
+  }
+
+  protected SinkRecord createKafkaRecord(String jsonString, boolean withSchema) {
```

Comment:
`withSchema` is always false. Do we need this argument?

Reply:
No it's not. I changed it.